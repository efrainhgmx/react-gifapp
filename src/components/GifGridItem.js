import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, url, title } ) => {

    console.log( id, url, title )
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propsTypes = {
    id: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
