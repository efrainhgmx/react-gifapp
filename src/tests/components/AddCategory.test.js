import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe('Test on <AddCategory /> component', () => {

    //Jest fn es una funcion falsa para pruebas
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    //Ciclo de vida de pruebas
    beforeEach( () => {

        //Limpia si hubo alguna simulacion en pruebas anteriores.
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });
    
    test('This component should be show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should be change de text box', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        const valueToCheck = wrapper.find('p').text().trim();
        input.simulate('change', { target: { value } });
        console.log(valueToCheck);

        expect( "Hola mundo" ).toBe( value );
    });

    test('Shouldnt show the change info with submit', () => {

       wrapper.find('form').simulate('submit', { preventDefault(){}});
        
       expect( setCategories ).not.toHaveBeenCalled();
    });

    test('Should call setCategories and clean the text box', () => {
        
        const form = wrapper.find('form');
        const input = wrapper.find('input');
        const value = 'One punch';
        
        //1.-Simular el inputChange
        input.simulate('change', { target: { value } });

        //2.- Simular el submit
        form.simulate('submit', { preventDefault(){}});

        //3.- setCategories debe haberse llamado
        //expect( setCategories ).toHaveBeenCalled();

        //4.- El valor del input debe estar ''
        expect(input.prop('value')).toBe("");
    })
    
    
    
    
})
