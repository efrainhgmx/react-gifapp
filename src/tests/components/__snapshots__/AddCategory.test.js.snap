// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AddCategory /> component This component should be show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
